File: src/main/kotlin/com/io/reactivecoroutines/WebClientConfig.java
package com.io.reactivecoroutines;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.client.reactive.ReactorClientHttpConnector;
import org.springframework.http.client.reactive.ReactorResourceFactory;
import org.springframework.web.reactive.function.client.WebClient;

import java.time.Duration;

@Configuration
public class WebClientConfig {
  @Bean
  public ReactorResourceFactory resourceFactory() {
    return new ReactorResourceFactory();
  }

  @Bean
  public WebClient webClient() {
    return WebClient.builder()
      .clientConnector(new ReactorClientHttpConnector(resourceFactory(), client -> client.responseTimeout(Duration.ofSeconds(10))))
      .build();
  }
}

File: src/main/kotlin/com/io/reactivecoroutines/WebClientConfig.kt
package com.io.reactivecoroutines

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.client.reactive.ReactorClientHttpConnector
import org.springframework.http.client.reactive.ReactorResourceFactory
import org.springframework.web.reactive.function.client.WebClient

import java.time.Duration

@Configuration
class WebClientConfig {
    @Bean
    fun resourceFactory(): ReactorResourceFactory = ReactorResourceFactory()

    @Bean
    fun webClient(): WebClient {
        return WebClient.builder()
            .clientConnector(ReactorClientHttpConnector(resourceFactory()) { client ->
                client.responseTimeout(Duration.ofSeconds(10))
            })
            .build()
    }
}